File: core/src/main/java/ar/com/wolox/wolmo/core/util/KeyboardUtils.java
/*
 * Copyright (c) Wolox S.A
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package ar.com.wolox.wolmo.core.util;

import android.app.Activity;
import android.content.Context;
import androidx.annotation.NonNull;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

import androidx.fragment.app.Fragment;

/**
 * An utility class to work with Android's keyboard
 */
public class KeyboardUtils {

    private KeyboardUtils() {}

    /**
     * Forces the device's soft keyboard (that means the software keyboard, not a physical one)
     * to show for a specific {@link EditText}
     *
     * @param context  an instance of {@link Context}
     * @param editText an instance of an attached {@link EditText}
     */
    public static void showKeyboard(@NonNull Context context, @NonNull EditText editText) {
        InputMethodManager imm =
                (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT);
    }

    /**
     * Forces the device's soft keyboard (that means the software keyboard, not a physical one)
     * to hide, meant to be called from inside a {@link Fragment}
     *
     * @param context an instance of {@link Context}
     * @param view    an instance of {@link Fragment
     */
    public static void hideKeyboard(@NonNull Context context, @NonNull View view) {
        InputMethodManager imm = (InputMethodManager) context
                .getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }

    /**
     * Forces the device's soft keyboard (that means the software keyboard, not a physical one)
     * to hide, meant to be called from inside an {@link Activity}
     *
     * @param activity an instance of the currently displayed {@link Activity}
     */
    public static void hideKeyboard(@NonNull Activity activity) {
        InputMethodManager imm = (InputMethodManager) activity.
                getSystemService(Activity.INPUT_METHOD_SERVICE);
        View view = activity.getCurrentFocus();
        // If no view currently has focus, create a new one,
        // just so we can grab a window token from it
        if (view == null) {
            view = new View(activity);
        }
        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }
}

File: core/src/main/java/ar/com/wolox/wolmo/core/util/KeyboardUtils.kt
package ar.com.wolox.wolmo.core.util

import android.app.Activity
import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import androidx.fragment.app.Fragment

class KeyboardUtils {

    companion object {
        /**
         * Forces the device's soft keyboard (that means the software keyboard, not a physical one)
         * to show for a specific [EditText]
         *
         * @param context  an instance of [Context]
         * @param editText an instance of an attached [EditText]
         */
        fun showKeyboard(context: Context, editText: EditText) {
            val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT)
        }

        /**
         * Forces the device's soft keyboard (that means the software keyboard, not a physical one)
         * to hide, meant to be called from inside a [Fragment]
         *
         * @param context an instance of [Context]
         * @param view    an instance of [Fragment]
         */
        fun hideKeyboard(context: Context, view: View) {
            val imm = context
                    .getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }

        /**
         * Forces the device's soft keyboard (that means the software keyboard, not a physical one)
         * to hide, meant to be called from inside an [Activity]
         *
         * @param activity an instance of the currently displayed [Activity]
         */
        fun hideKeyboard(activity: Activity) {
            val imm = activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
            var view = activity.currentFocus
            // If no view currently has focus, create a new one,
            // just so we can grab a window token from it
            if (view == null) {
                view = View(activity)
            }
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }
}